packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -222,31 +239,35 @@ export class PipelineEditor extends React.Component<
       {
         action: 'export',
         label: 'Export Pipeline',
-        enable: true,
+        enable: !this.state.emptyPipeline,
         iconEnabled: IconUtil.encode(exportPipelineIcon),
         iconDisabled: IconUtil.encode(exportPipelineIcon)
       },
       {
         action: 'clear',
         label: 'Clear Pipeline',
-        enable: true,
+        enable: !this.state.emptyPipeline || !emptyCanvas,
         iconEnabled: IconUtil.encode(clearPipelineIcon),
         iconDisabled: IconUtil.encode(clearPipelineIcon)
       },
       { divider: true },
-      { action: 'undo', label: 'Undo', enable: true },
-      { action: 'redo', label: 'Redo', enable: true },
-      { action: 'cut', label: 'Cut', enable: false },
-      { action: 'copy', label: 'Copy', enable: false },
-      { action: 'paste', label: 'Paste', enable: false },
+      { action: 'undo', label: 'Undo' },
+      { action: 'redo', label: 'Redo' },
+      { action: 'cut', label: 'Cut' },
+      { action: 'copy', label: 'Copy' },
+      { action: 'paste', label: 'Paste' },
       { action: 'addComment', label: 'Add Comment', enable: true },
-      { action: 'delete', label: 'Delete', enable: true },
+      { action: 'delete', label: 'Delete' },
```

Comment:
We don't need to set the `enable` property on these commands, they're handled by context on canvas already